validator/client/runner.go

```diff
@@ -129,8 +130,9 @@ func run(ctx context.Context, v iface.Validator) {
 
 			if slots.IsEpochStart(slot) && v.ProposerSettings() != nil {
 				go func() {
-					//deadline set for next epoch rounded up
-					if err := v.PushProposerSettings(ctx, km); err != nil {
+					//deadline set for end of epoch
+					epochDeadline := v.SlotDeadline(slot + params.BeaconConfig().SlotsPerEpoch - 1)
```

Comment:
This doesn't seem like the end of the epoch, if I am understanding this correctly. If our current slot is 20 , then we would use 51 as the slot for the deadline. However the end of the current epoch would be at the end of slot 31 and not 51. As 51 comes under the next epoch

Reply:
it shouldn't run on slot 20 as the if statement above checks if the slot is epoch start. hope I'm thinking about that correctly..